controllers/costmanagement_controller.go

```diff
@@ -132,12 +158,104 @@ func GetClusterID(r *CostManagementReconciler, cost *costmgmtv1alpha1.CostManage
 	return nil
 }
 
+func GetPullSecretToken(r *CostManagementReconciler, costInput *CostManagementInput) error {
+	ctx := context.Background()
+	log := r.Log.WithValues("costmanagement", "GetPullSecretToken")
+	secret := &corev1.Secret{}
+	namespace := types.NamespacedName{
+		Namespace: openShiftConfigNamespace,
+		Name:      pullSecretName}
+	err := r.Get(ctx, namespace, secret)
+	if err != nil {
+		switch {
+		case errors.IsNotFound(err):
+			log.Error(err, "Pull-secret does not exist.")
+		case errors.IsForbidden(err):
+			log.Error(err, "Operator does not have permission to check pull-secret.")
+		default:
+			log.Error(err, "Could not check pull-secret.")
+		}
+		return err
+	}
+
+	tokenFound := false
+	encodedPullSecret := secret.Data[pullSecretDataKey]
+	if len(encodedPullSecret) <= 0 {
+		return fmt.Errorf("Cluster authorization secret didn't have data.")
+	}
+	var pullSecret serializedAuthMap
+	if err := json.Unmarshal(encodedPullSecret, &pullSecret); err != nil {
+		log.Error(err, "Unable to unmarshal cluster pull-secret.")
+		return err
+	}
+	if auth, ok := pullSecret.Auths[pullSecretAuthKey]; ok {
+		token := strings.TrimSpace(auth.Auth)
+		if strings.Contains(token, "\n") || strings.Contains(token, "\r") {
+			return fmt.Errorf("Cluster authorization token is not valid: contains newlines.")
+		}
+		if len(token) > 0 {
+			log.Info("Found cloud.openshift.com token.")
+			costInput.BearerTokenString = token
+			tokenFound = true
+		} else {
+			return fmt.Errorf("Cluster authorization token is not found.")
+		}
+	} else {
+		return fmt.Errorf("Cluster authorization token was not found in secret data.")
+	}
+	if !tokenFound {
+		return fmt.Errorf("Cluster authorization token is not found.")
+	}
+	return nil
+}
+
+func GetAuthSecret(r *CostManagementReconciler, costInput *CostManagementInput, reqNamespace types.NamespacedName) error {
+	ctx := context.Background()
+	log := r.Log.WithValues("costmanagement", "GetAuthSecret")
+
+	log.Info("Secret namespace", "namespace", reqNamespace.Namespace)
+	secret := &corev1.Secret{}
+	namespace := types.NamespacedName{
+		Namespace: reqNamespace.Namespace,
+		Name:      costInput.AuthenticationSecretName}
+	err := r.Get(ctx, namespace, secret)
+	if err != nil {
+		switch {
+		case errors.IsNotFound(err):
+			log.Error(err, "Secret does not exist.")
```

Comment:
Logging the error here still shows the traceback logs - but I'll leave it up to you whether you want to quiet them all or not.

Reply:
I was going to leave the initial. You all can circle back if you prefer to quiet them all.